docker/Dockerfile

```diff
@@ -1,102 +1,90 @@
-FROM python:3.9-slim-buster
-
-LABEL maintainer="Eddie Antonio Santos <Eddie.Santos@nrc-cnrc.gc.ca>"
-
-# (2021-01-15): This Dockerfile was derived from Gūnáhà:
-#  - https://github.com/UAlbertaALTLab/gunaha/blob/master/Dockerfile
+# = CHANGELOG =
+#
+# 2021-02-19:
 #
-# Changes:
+#  - Use a multi-stage build to keep the application image small
 #
+# 2021-01-15:
+#
+#  - This Dockerfile was derived from Gūnáhà:
+#    https://github.com/UAlbertaALTLab/gunaha/blob/master/Dockerfile
 #  - use Gunicorn instead of uwsgi; I'm not sure why I chose uwsgi for Gūnáhà,
 #    but I've got itwêwina working with Gunicorn, so I'm gonna stick with that
 #
 
+################################ Build stage #################################
+FROM python:3.9-slim-buster AS builder
 
-# Directories:
-#
-# /app — the app code will live here
-#
+LABEL maintainer="Eddie Antonio Santos <Eddie.Santos@nrc-cnrc.gc.ca>"
 
-# Choose an ID that will be consistent across all machines in the network
-# To avoid overlap with user IDs, use an ID over
-# /etc/login.defs:/UID_MAX/, which defaults to 60,000
 ARG UID_GID=60003
-ARG EXPOSE_PORT=8000
+ARG NODE_VERSION=v12.20.2
+
+# Install Node + npm
+# (note: default npm is too old for the apt install'd version of NodeJS??!?!?!)
+WORKDIR /tmp
+ADD https://nodejs.org/dist/${NODE_VERSION}/node-${NODE_VERSION}-linux-x64.tar.gz /tmp/node.tar.gz
+RUN tar xf node.tar.gz \
+ && mv node-${NODE_VERSION}-linux-x64 /opt/node
+ENV PATH="/opt/node/bin:${PATH}"
```

Comment:
Oh cool, I did not realize you could reference `$PATH` when resetting it in a Dockerfile and was embedding the default …

Reply:
I learned this too today. I _considered_ copying everything into `/usr/local` but this seemed much safer!